src/Main.elm

```diff
@@ -1,10 +1,829 @@
 module Main exposing (main)
 
+import Browser
+import Browser.Events
 import Data.Audience
 import Data.AudienceFolder
-import Html exposing (Html)
+import Dict
+import Element as E -- elm-ui
+import Element.Background as EBackground
+import Element.Border as EBorder
+import Element.Events as EEvents
+import Element.Font as EFont
+import Element.Input as EInput
+import Html
+import Html.Attributes
+import Icons
+import Json.Decode
+import Palette
+import Process
+import Task
+import Time
 
 
-main : Html msg
+
+-- we could perhaps model this so that you can only make a "FolderClicked" Msg
+-- or "SortBy" Msg only after the model has loaded up, but I don't think this is
+-- too much of an issue
```

Comment:
Yeah, same here. I usually tend to stray away from really nested Msgs because then I usually find it's harder to trace back where something came from. Whereas if it's just a more flat `Msg` type, I find it easier to find stuff in one single file.